Intersect.Server/Entities/ProjectileSpawn.cs

```diff
@@ -122,16 +118,16 @@ public bool HitEntity(Entity targetEntity)
                         }
                     }
                 }
-                else if (targetEntity is Resource)
+                else if (targetEntity is Resource targetResource)
                 {
-                    if (((Resource) targetEntity).IsDead() && !ProjectileBase.IgnoreExhaustedResources ||
-                        !((Resource) targetEntity).IsDead() && !ProjectileBase.IgnoreActiveResources)
+                    if (targetResource.IsDead() && !ProjectileBase.IgnoreExhaustedResources ||
+                        !targetResource.IsDead() && !ProjectileBase.IgnoreActiveResources)
                     {
-                        if (Parent.Owner is Player && !((Resource) targetEntity).IsDead())
+                        if (Parent.Owner is Player && !targetResource.IsDead())
                         {
-                            Parent.Owner.TryAttack(targetEntity, Parent.Base, Parent.Spell, Parent.Item, Dir);
+                            Parent.Owner.TryAttack(targetResource, Parent.Base, Parent.Spell, Parent.Item, Dir);
 
-                            if (Dir <= 3 && ShouldHook(targetEntity) && !Parent.HasGrappled)
+                            if (Dir <= 3 && ShouldHook(targetResource) && !Parent.HasGrappled)
```

Comment:
I think this can be simplified by just allowing multiple `return true;`s.

We're checking `IsDead()` 3 times here when realistically we should check once.

Pseudocode:

```
if (dead) {
  if (!IgnoreExhaustedResources) {
    return true;
  }
} else if (!IgnoreActiveResources) {
  if (Parent.Owner is Player) {
    ... tryattack ...
    ... if dir <= 3 ...
    
  }
  
  return true;
}